kernel/kernel-api/src/test/scala/io/delta/kernel/internal/icebergcompat/IcebergCompatV2MetadataValidatorAndUpdaterSuite.scala

```diff
@@ -95,19 +109,78 @@ class IcebergCompatV2MetadataValidatorAndUpdaterSuite
   Seq(true, false).foreach { isNewTable =>
     test(s"can't be enabled on a table with deletion vectors supported, isNewTable $isNewTable") {
       val schema = new StructType().add("col", BooleanType.BOOLEAN)
-      val metadata = testMetadata(schema, Seq.empty).withIcebergCompatV2AndCMEnabled()
-      val protocol = testProtocol(
-        ICEBERG_COMPAT_V2_W_FEATURE,
-        COLUMN_MAPPING_RW_FEATURE,
-        DELETION_VECTORS_RW_FEATURE)
+      val metadata = getCompatEnabledMetadata(schema)
+      val protocol = getCompatEnabledProtocol(DELETION_VECTORS_RW_FEATURE)
       val e = intercept[KernelException] {
-        validateAndUpdateIcebergCompatV2Metadata(isNewTable, metadata, protocol)
+        runValidateAndUpdateIcebergCompatV2Metadata(isNewTable, metadata, protocol)
       }
       assert(e.getMessage.contains(
         "Simultaneous support for icebergCompatV2 and deletion vectors is not compatible."))
     }
   }
 
+  Seq(true, false).foreach { isNewTable =>
```

Comment:
These two are just moved/refactored from lower in the test suite (for some reason the diff recognized the other ones but not these)